cellprofiler/__main__.py

```diff
@@ -575,18 +587,23 @@ def get_batch_commands(filename):
                 print "CellProfiler -c -r -p %s -f %d -l %d" % (filename, prev + 1, off)
 
                 prev = off
+    else:
+        metadata_tags = m.get_grouping_tags()
 
-            return
-
-    metadata_tags = m.get_grouping_tags()
-
-    groupings = m.get_groupings(metadata_tags)
-
-    for grouping in groupings:
-        group_string = ",".join(["%s=%s" % (k, v) for k, v in grouping[0].iteritems()])
+        if len(metadata_tags) == 1 and metadata_tags[0] == 'ImageNumber':
+            for i in range(0, len(image_numbers), n_per_job):
+                first = image_numbers[i]
+                last = image_numbers[min(i+n_per_job-1, len(image_numbers)-1)]
+                print "CellProfiler -c -r -p %s -f %d -l %d" % (filename, first, last)
+        else:
+            # LoadData w/ images grouped by metadata tags
```

Comment:
I re-read https://www.python.org/dev/peps/pep-0008/#blank-lines. We do adhere to

> Surround top-level function and class definitions with two blank lines.
>
> Method definitions inside a class are surrounded by a single blank line.  

These are the most well-defined PEP08 style recommendations on blank lines.

> Use blank lines in functions, sparingly, to indicate logical sections. 

I suppose we could debate about what defines a "logical section", and I choose to leave that decision up to the contributor. It is my opinion that functions should be simple enough to accomplish one task. A function is a sequence of operations. Generally, each operation is its own logical section and rarely do these logical units span more than one line. If there are many clumps, or logical sections, then it's a good indication refactoring may be needed.